registry/storage/blobstore.go

```diff
@@ -85,6 +87,52 @@ func (bs *blobStore) Put(ctx context.Context, mediaType string, p []byte) (distr
 	}, bs.driver.PutContent(ctx, bp, p)
 }
 
+func (bs *blobStore) Create(ctx context.Context, options ...distribution.BlobCreateOption) (distribution.BlobWriter, error) {
```

Comment:
BlobEnumerator should be enough although BlobService has the nice property of also being a BlobStatter which is useful for my unit tests for constructing a manifest list. Would it be reasonable to create a Blob***** type in blobs.go that represents the ability to read and iterate through blobs?